internal/sync/syncer.go

```diff
@@ -239,6 +245,18 @@ func (s *Syncer) runOnce(ctx context.Context, key string, worker int) error {
 		Namespace: gitSync.Namespace,
 		Name:      gitSync.Name,
 	}
+
+	if !gitSync.GetDeletionTimestamp().IsZero() {
```

Comment:
This should be the last thing during sync task and after we do `updateCommitStatus`.